c7n/resources/mq.py

```diff
@@ -83,10 +85,54 @@ def get_dimensions(self, resource):
                  'Value': "{}-1".format(resource['BrokerName'])}]
 
 
+@MessageBroker.filter_registry.register('vpc')
+class VpcFilter(VpcFilter):
+    """Filter a resource by its VPC id. Choose to select resources that are in that VPC or select
+    resources that are not in that VPC using the 'op' field.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: mq-vpc-filters
+                resource: message-broker
+                filters:
+                      - type: vpc
+                        key: VpcId
+                        value: vpc-xxxxxxxxxxx
+                        op: eq
+    """
+
+    def get_related_ids(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
```

Comment:
this method gets called quite a bit in related resource processing, ie once for the set, and potentially once per resource. any api call usage should be cached.